### ISSUE TITLE
Status checks do not time out

### ISSUE THREAD

**Issue by Smiley73** [KohlsTechnology/eunomia#241]
**What version of eunomia are you using?**
eunomia version: 0.1.0

**Does this issue reproduce with the latest release?**

**What operating system and processor architecture are you using (`kubectl version`)?**
<details><summary><code>Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0+d4cacc0", GitCommit:"d4cacc0", GitTreeState:"clean", BuildDate:"2019-03-19T15:41:07Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}</code> Output</summary><br><pre>
$ kubectl version
</pre></details>

**What did you do?**
1. apply CR (cronjob & change trigger)
2. let it do it’s thing and wait a bit
3. delete CR
4. let it clean up everything and wait for all pods to finish
5. reapply CR and it’s stuck. The eunomia log shows it’s still waiting for an old job that doesn’t exist anymore 
6. restart eunomia and it'll process the CR correctly

**What did you expect to see?**
I can create & delete the same CR over and over in the same namespace, without eunomia getting stuck.

**What did you see instead?**
The below message shows up over and over in the eunomia log, until I restart.
It seems the status check never times out.
```
{"level":"info","ts":1578692721.2975926,"logger":"gitopsconfig-controller","msg":"retrying the GetJob for CheckResult"}
{"level":"info","ts":1578692721.297719,"logger":"gitopsconfig-controller","msg":"Retry of GetJob for CheckResult unsuccessful: Job.batch \"gitopsconfig-eunomia-test-ace7az\" not found"}
```

**Comment by seanmalloy** [comment]
Might be handled by #217

**Comment by seanmalloy** [comment]
Fixed by #249.
